src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,135 @@
+    }
+
+    /**
+     * Prints when users asks for the list of tasks.
+     *
+     * @param list the list of tasks.
+     */
+    public void printList(TaskList list) {
+        if (list.getSize() == 0) {
+            System.out.println("There are no tasks in your list.");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < list.getSize(); i++) {
+                System.out.println((i + 1) + ". " + list.getTask(i).toString());
+            }
+        }
+    }
+
+    public void printMatchingTasks(ArrayList<Task> matchingTasks) {
+        System.out.println("Here are the matching tasks in your list:");
+        for (int i = 0; i < matchingTasks.size(); i++) {
+            System.out.println((i + 1) + ". " + matchingTasks.get(i).toString());
+        }
+
+    }
+
+    /**
+     * Prints when user marks a task as done.
+     *
+     * @param list the list of tasks.
+     * @param number the task number to be marked as done.
+     */
+    public void mark(TaskList list, int number) {
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(list.getTask(number).toString());
+    }
+
+    /**
+     * Prints when user unmarks a done task back to not done yet.
+     *
+     * @param list the list of tasks.
+     * @param number the task number to be unmarked.
+     */
+    public void unMark(TaskList list, int number) {
+        System.out.println("OK, I've marked this task as not done yet:");
+        System.out.println(list.getTask(number).toString());
+    }
+
+    /**
+     * Prints when user wants to add a todo to the task list.
+     *
+     * @param list the list of tasks.
+     * @param newTodo the new todo to be added.
+     */
+    public void addTodo(TaskList list, Todo newTodo) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println(newTodo.toString());
+        System.out.println("Now you have " + list.getSize() + " tasks in the list.");
+    }
+
+    /**
+     * Prints when user wants to add a deadline to task list.
+     *
+     * @param list the list of tasks.
+     * @param newDeadline the new deadline to be added.
+     */
+    public void addDeadline(TaskList list, Deadline newDeadline) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println(newDeadline.toString());
+        System.out.println("Now you have " + list.getSize() + " tasks in the list.");
+    }
+
+    /**
+     * Prints when user wants to add an event to task list.
+     *
+     * @param list the list of tasks.
+     * @param newEvent the new event to be added.
+     */
+    public void addEvent(TaskList list, Event newEvent) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println(newEvent.toString());
+        System.out.println("Now you have " + list.getSize() + " tasks in the list.");
+    }
+
```

Comment:
In general, I think that you have done quite well in documentation, naming of methods and variable names, naming seem intuitive and well ordered, well done! 👍 😄